app/redux/Home/HomeActions.tsx

```diff
@@ -0,0 +1,24 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { AUTH_KEY, baseURL } from '../../services/ApiConfig';
+import { ApiErrorType, ApiResponseType } from '../../services/Types';
+import type { TrendingApiRequestParams } from './HomeTypes';
+
+export const getTrendingData = createAsyncThunk<
+  ApiResponseType,
+  TrendingApiRequestParams,
+  { rejectValue: ApiErrorType }
+>('trending', async (trendingData: TrendingApiRequestParams, thunkApi) => {
+  const { time, type } = trendingData;
+  try {
+    let response = await fetch(
+      baseURL + 'trending/' + type + '/' + time + '?api_key=' + AUTH_KEY,
+    );
+    let json = await response.json();
+    return json;
+  } catch (err: any) {
+    const error: string = err?.response?.data?.error?.description;
+    return thunkApi.rejectWithValue({
+      status_message: error ?? 'Error',
+    } as ApiErrorType);
```

Comment:
used class instead of type for ApiErrorType and here used thunkApi.rejectWithValue(new ApiErrorType(error ?? 'Error'));